routing/pathfind_test.go

```diff
@@ -3327,3 +3434,156 @@ func TestBlindedRouteConstruction(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, expectedRoute, route)
 }
+
+// TestLastHopPayloadSize tests the final hop payload size. The final hop
+// payload structure differes from the intermediate hop payload for both the
+// non-blinded and blinded case.
+func TestLastHopPayloadSize(t *testing.T) {
```

Comment:
Regarding this test I am not sure if this test is independent enough wdyt? I have the feeling I am just reengineering the tests of `route.Hop` because I am using the `Payload` creation function to build the payload package and then comparing it to the size function which is basically the wrapped `PayloadSize` function of `route.Hop`

Reply:
I tried that now, but for this we should also move the `LastPayloadSize` into the `route` pkg (I guess?), but then I get a circular dependency regarding the `RestrictParams` argument. So I think it's fine for now to leave it here?